src/app/work-item/work-item.service.ts

```diff
@@ -40,21 +41,17 @@ export class WorkItemService {
       .catch(this.handleError);
   }
 
-  getWorkItemTypes(): Promise<any[]> { 
-    if (this.workItemTypes.length) {
-      return new Promise((resolve, reject) => {
-        resolve(this.workItemTypes);
-      });
-    } else {
-      return this.http
-      .get(this.workItemTypeUrl)
+ getWorkItemTypes(): Promise<WorkItemType[]> {
+    this.logger.log('WorkItemService using url ' + this.workItemTypeUrl);
+    return this.http
+      .get(this.workItemTypeUrl + '.2', {headers: this.headers})
```

Comment:
I think the point of having a variable for the URL is so that you don't have to change it in all the places it is used. Adding '.2' is the type of change this is meant to avoid.